### PR TITLE
Large UX redesign to align app with wireframes, and to update resource-type selection

### PR DESCRIPTION
**Pull request by sleitner**

### CODE REVIEW

File: ckanext/cfpb_extrafields/datastore_actions.py [hunk 1]
```diff
@@ -10,52 +10,83 @@
...
+        'force':'true',
+        'fields': defaults.all_fields,
+        'records': []
     })
-    ds = tk.get_action('datastore_create')(context, data)
-def get_all_datastore_jsons(rid, title_colname, json_title): 
+    ds = tk.get_action('datastore_create')(defaults.context, data)
+
+# JSON related (for datadict):
+def create_datastore_json(rid, json_title, json_record):
```
Comment (tanderegg) [inline ckanext/cfpb_extrafields/datastore_actions.py]: Possibly merge this method with create_datastore_placeholder and use default values of "None" for json_title and json_record.

File: ckanext/cfpb_extrafields/datastore_actions.py [hunk 2]
```diff
@@ -10,52 +10,83 @@
...
-def create_datastore_json(rid, json_record, title_colname, json_title): 
-    ''' create a new datastore json element with primary key element of title_colname and record '''
+    defaults_tuple = collections.namedtuple('defaults_tuple', 'context data \
+    title_colname subtitle_colname json_colname datetime_colname response_colname \
+    response_timems_colname all_fields')
+    return defaults_tuple(context, data,
+                          title_colname, subtitle_colname, json_colname,
+                          datetime_colname, response_colname,
+                          response_timems_colname, all_fields)
+def create_datastore_placeholder(rid):
```
Comment (tanderegg) [inline ckanext/cfpb_extrafields/datastore_actions.py]: Add two line breaks to comply with PEP8

File: ckanext/cfpb_extrafields/datastore_actions.py [hunk 3]
```diff
@@ -10,52 +10,83 @@
...
+    ''' create a new datastore json element with title_colname and record '''
+    defaults = default_datastore(rid)
+    data = defaults.data
+    data.update({
+        'force':'true',
+        'fields': defaults.all_fields,
+        'records': [ {defaults.title_colname: json_title, defaults.json_colname: json_record} ]
+    })
+    ds = tk.get_action('datastore_create')(defaults.context, data)
+def get_all_datastore_jsons(rid, json_title):
```
Comment (tanderegg) [inline ckanext/cfpb_extrafields/datastore_actions.py:56]: 2 line breaks for PEP8

File: ckanext/cfpb_extrafields/fanstatic/listrelated.js [hunk 4]
```diff
@@ -27,7 +27,35 @@ ckan.module('listrelated', function ($, _) {
...
+                if ($(this).data('gist-url')) {
+                    // Create an iframe, append it to this document where specified
+                    var gistFrame = document.createElement("iframe");
+                    gistFrame.setAttribute("width", "100%");
+                    gistFrame.id = "gistFrame";
+
+                    $(this).find('.expandable_content').html(gistFrame);
+                    
+                    // Create the iframe's document
+                    var gistFrameHTML = '<html><body style="margin:0"><scr'+'ipt type="text/javascript" src="' + $(this).data('gist-url') + '.js"></sc'+'ript></b'+'ody></h'+'tml><base target="_parent" />';
```
Comment (tanderegg) [inline ckanext/cfpb_extrafields/fanstatic/listrelated.js:41]: Just check weather extra concatenations are necessary.

File: ckanext/cfpb_extrafields/plugin.py [hunk 5]
```diff
@@ -51,30 +51,29 @@ def parse_resource_related_gist(data_related_items, resource_id):
...
 
     p.implements(p.IResourceController)
-    def before_create(self, context, resource):
-        return
-    def after_create(self, context, resource):
-        # must have changed['format'] when you created the resource
-        tk.redirect_to(controller='package',action='resource_edit',
-                       resource_id=resource['id'],id=resource['package_id'])
-
+    def _which_check_keys_changed(self, old, new):
```
Comment (tanderegg) [inline ckanext/cfpb_extrafields/plugin.py:58]: Single line breaks for methods within a class

File: ckanext/cfpb_extrafields/datastore_actions.py (hunk 1 above)
Comment (sleitner) [inline ckanext/cfpb_extrafields/datastore_actions.py]: :+1: good call

File: ckanext/cfpb_extrafields/plugin.py [hunk 6]
```diff
@@ -51,30 +51,29 @@ def parse_resource_related_gist(data_related_items, resource_id):
...
         import json
-        if 'datadict' in resource:
+        if 'datadict' in resource and 'id' in resource:
             record = resource['datadict']
             resource.pop('datadict')
             json_record = json.loads(record)
             try:
-                ds.delete_datastore_json(resource['id'], 'json_name', 'datadict')
+                ds.delete_datastore_json(resource['id'], 'datadict')
             except tk.ObjectNotFound, err:
```
Comment (tanderegg) [inline ckanext/cfpb_extrafields/plugin.py]: Can check for multiple errors in one line (error1, error2)

File: ckanext/cfpb_extrafields/templates/package/resource_read.html [hunk 7]
```diff
@@ -1,200 +1,351 @@
...
-    {% if res.format == 'Data Dictionary'  %}
-      {% snippet 'package/snippets/format-specific/cfpb_extrafields_datadictionary.html',data=res,errors=errors,pkg_name=res.name,editable="false" %}
-    {% endif %}
+{% block resource_content %}
+
+{% block resource_read_title %}
+<h1 class="page-heading">{{ h.resource_display_name(res) | truncate(50) }}</h1>
+{% endblock %}
+{% block resource_read_url %}
+{# URL is can also be seen in Go to Resource and this is distracting #}
```
Comment (tanderegg) [inline ckanext/cfpb_extrafields/templates/package/resource_read.html]: Edit for grammar

**Comment by Ooblioob** [comment]
@tanderegg thanks for your comments on the PR!  Great work @sleitner!

**Pull request merged**